micro/server.py

```diff
@@ -965,6 +966,36 @@ def post(self) -> None:
         self.set_status(HTTPStatus.CREATED) # type: ignore
         self.write(referral.json(restricted=True, include=True))
 
+    def get(self) -> None:
+        """ Return a summary of referrals over a period of time.
+
+            Query parameters:
+
+            * `start` an iso timestamp or date representing the start of the period of interest
+            * `end` an iso timestamp or date representing the end of the period of interest
+
+            If `start` or `end` are missing, the period will default to [now - 7 days, now]
+
+            Returns a JSON representation of the result created by analytics.Referrals.summarize
+        """
+        start = self.get_query_argument("start", None)
+        end = self.get_query_argument("end", None)
+
+        if start is None or end is None:
+            end = self.app.now()
+            start = end - timedelta(days=7)
+        else:
+            try:
+                end = datetime.fromisoformat(end)
+                start = datetime.fromisoformat(start)
+            except ValueError:
+                raise HTTPError(HTTPStatus.BAD_REQUEST)
```

Comment:
We should raise an `error.ValueError('Bad start or end')`, which gets translated by the server to a `BAD REQUEST` with a JSON representation.